main.go

```diff
@@ -119,18 +142,31 @@ func markPDF(inputPath string, outputPath string, watermark string) error {
 			drawImage(watermarkImg, c)
 
 		} else {
-			if pageNum == 1 {
-				para = creator.NewParagraph(watermark)
-				adjustTextPosition(para, c)
-			}
+
+			// Execute the template for each page.
+			buf.Reset()
+			err := t.Execute(buf, rec)
+			fatalIfError(err, fmt.Sprintf("Failed to execute watermark template: [%s]", err))
+
+			para = creator.NewParagraph(buf.String())
+			adjustTextPosition(para, c)
 
 			drawText(para, c)
 		}
 
 	}
 
-	err = c.WriteToFile(outputPath)
-	return err
+	// Create a new template and parse the output path into it.
+	oP := template.Must(template.New("outputPath").Parse(outputPath))
+	buf.Reset()
+	err = oP.Execute(buf, rec)
+	fatalIfError(err, fmt.Sprintf("Failed to execute output path template: [%s]", err))
+
+	err = c.WriteToFile(buf.String())
+	fatalIfError(err, fmt.Sprintf("Unable to Write file: [%s]", err))
```

Comment:
This whole section can be moved to a separate function like `writeToFileWithTemplate(outputPath, other...)` (just a suggestion) :) 

And, as mentioned in previous comment, can be used `c.WriteToFile(outputPath)` or `writeToFileWithTemplate ` based on placeholders are used or not.